app/client/src/components/room/SingleRoom.vue

```diff
@@ -12,12 +12,21 @@ const props = defineProps({
     },
 });
 
-// Whether has unread
 const hasUnread = computed(() => {
     return !selected.value && client.hasAccessToRoom(props.room.id) && client.hasUnreadMessages(props.room.id);
 });
 
-// Whether this room is selected
+const isMuted = ref(false);
+watch(
+    client.state,
+    () => {
+        const data = apiClient.plugins.mute.isRoomMuted(props.room.id);
+        isMuted.value = data;
+    },
+    { deep: true },
+);
```

Comment:
Unfortunately `apiClient.plugins.mute.isRoomMuted` isn't accessing reactive data, so we must do this for now.

There should be a wrapper that listens on `client.state` but behaves as `computed(..)`. Not optimal performance-wise but would clean the code in quite some places

Reply:
Updated to use a composable that can centralize this watching logic. Performance wise, can be improved (since many callbacks will watch with `{deep: true}` on the API global state object. But at least we have a safe way forward and it'll be easier to upgrade with this composable.